pkg/server/api/agent/v1/service.go

```diff
@@ -47,7 +55,52 @@ func (s *Service) DeleteAgent(ctx context.Context, req *agent.DeleteAgentRequest
 }
 
 func (s *Service) BanAgent(ctx context.Context, req *agent.BanAgentRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "method not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if req.Id == nil {
+		log.Error("Invalid request: ID is nil")
+		return nil, status.Error(codes.InvalidArgument, "ID is nil")
+	}
+
+	id, err := spiffeid.New(req.Id.TrustDomain, req.Id.Path)
+	if err != nil {
+		log.WithError(err).Error("Invalid request: invalid SPIFFE ID")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid SPIFFE ID: %v", err)
+	}
+
+	log = log.WithField(telemetry.SPIFFEID, id.String())
+
+	if !strings.Contains(id.Path(), "spire/agent/") {
+		log.Error("Invalid request: ID is not an agent SPIFFE ID")
+		return nil, status.Error(codes.InvalidArgument, "ID is not an agent SPIFFE ID")
+	}
+
+	if !id.MemberOf(s.td) {
+		log.Error("Invalid request: cannot ban an agent that does not belong to this trust domain")
+		return nil, status.Errorf(codes.InvalidArgument, "cannot ban an agent that does not belong to this trust domain")
+	}
+
+	_, err = s.ds.UpdateAttestedNode(ctx, &datastore.UpdateAttestedNodeRequest{
+		SpiffeId: id.String(),
+		InputMask: &common.AttestedNodeMask{
+			CertSerialNumber:    true,
```

Comment:
so it measn we set serials as emtpy to ban it, right? can you add comments about it?

Reply:
Yes, I'll add a comment about that 👍